client/blocks/upe/fields.js

```diff
@@ -0,0 +1,214 @@
+	const [ isUpeComplete, setIsUpeComplete ] = useState( false );
+	const [ errorMessage, setErrorMessage ] = useState( null );
+
+	const paymentMethodsConfig = getStripeServerData()?.paymentMethodsConfig;
+
+	useEffect( () => {
+		if ( paymentIntentId || hasRequestedIntent ) {
+			return;
+		}
+
+		async function createIntent() {
+			try {
+				const response = await api.createIntent(
+					getStripeServerData()?.orderId
+				);
+				setPaymentIntentId( response.id );
+				setClientSecret( response.client_secret );
+			} catch ( error ) {
+				setErrorMessage(
+					error?.message ??
+						'There was an error loading the payment gateway'
+				);
+			}
+		}
+
+		setHasRequestedIntent( true );
+		createIntent();
+	}, [ paymentIntentId, hasRequestedIntent, api, errorMessage ] );
+
+	useEffect(
+		() =>
+			onPaymentProcessing( () => {
+				if ( activePaymentMethod !== 'stripe' ) {
+					return;
+				}
+
+				if ( ! isUpeComplete ) {
+					return {
+						type: 'error',
+						message: 'Your payment information is incomplete.',
+					};
+				}
+
+				if ( errorMessage ) {
+					return {
+						type: 'error',
+						message: errorMessage,
+					};
+				}
+
+				if (
+					shouldSavePayment &&
+					! paymentMethodsConfig[ selectedUpePaymentType ].isReusable
+				) {
+					return {
+						type: 'error',
+						message:
+							'This payment method can not be saved for future use.',
+					};
+				}
+
+				return {
+					type: 'success',
+					meta: {
+						paymentMethodData: {
+							paymentMethod: PAYMENT_METHOD_NAME,
+							wc_payment_intent_id: paymentIntentId,
+						},
+					},
+				};
+			} ),
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+		[ activePaymentMethod, isUpeComplete, shouldSavePayment ]
+	);
+
+	useEffect(
+		() =>
+			onCheckoutAfterProcessingWithSuccess(
+				( { orderId, processingResponse: { paymentDetails } } ) => {
+					async function updateIntent() {
+						await api.updateIntent(
+							paymentIntentId,
+							orderId,
+							shouldSavePayment ? 'yes' : 'no',
+							selectedUpePaymentType
+						);
+
+						const paymentElement = elements.getElement(
+							PaymentElement
+						);
+
+						return confirmUpePayment(
+							api,
+							paymentDetails.redirect_url,
+							paymentDetails.payment_needed,
+							paymentElement,
+							billingData,
+							emitResponse
+						);
+					}
+
+					return updateIntent();
+				}
+			),
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+		[
+			api,
+			elements,
+			paymentIntentId,
+			selectedUpePaymentType,
+			shouldSavePayment,
+			stripe,
+		]
+	);
+
+	const elementOptions = {
+		clientSecret,
+		business: { name: getStripeServerData()?.accountDescriptor },
+		fields: {
+			billingDetails: {
+				name: 'never',
+				email: 'never',
+				phone: 'never',
+				address: {
+					country: 'never',
+					line1: 'never',
+					line2: 'never',
+					city: 'never',
+					state: 'never',
+					postalCode: 'never',
+				},
+			},
```

Comment:
I'm pretty sure these shouldn't all be set to `'never'`, but instead use the `enabledBillingFields` setting we embed from the PHP side. Not sure how that plays with the Blocks integration though; that's something we need to look into.